components/Variables/Var.tsx

```diff
@@ -0,0 +1,56 @@
+import cn from "classnames";
+import { useCallback, useContext, useEffect } from "react";
+import { VarsContext } from "./context";
+import styles from "./Var.module.css";
+
+interface VarProps {
+  name: string;
+  needLabel?: boolean;
+  isGlobal?: boolean;
+}
+
+export const Var = ({ name, needLabel, isGlobal }: VarProps) => {
+  const { fields, setField, markGlobalField, addField } =
+    useContext(VarsContext);
+
+  useEffect(() => {
+    if (isGlobal) {
+      markGlobalField(name);
+    } else {
+      addField(name);
+    }
+  }, [isGlobal, name]);
```

Comment:
Github is showing missing dependencies:
```
React Hook useEffect has missing dependencies: 'addField' and 'markGlobalField'.
```

Reply:
we do not add functions `markGlobalField` and `addField` to the dependency array, since they do not change (constant useCallback)

Reply:
You are getting a warning here because it is flagging these appropriately. `addField` and `markGlobalField` are from a context, which can update, and do, on every render. These two functions should definitely be added to the dependency array.

Reply:
I changed the functions a bit and now there is only one function to add a field. I added it to the dependency array